core/runtime/src/main/java/io/quarkus/runtime/util/ContainerRuntimeUtil.java

```diff
@@ -77,46 +77,48 @@ public static ContainerRuntime detectContainerRuntime(boolean required) {
             if (dockerAvailable) {
                 // Check if "docker" is an alias to "podman"
                 if (dockerVersionOutput.startsWith("podman version")) {
-                    storeConfig(ContainerRuntime.PODMAN);
+                    storeContainerRuntimeInSystemProperty(ContainerRuntime.PODMAN);
                     return ContainerRuntime.PODMAN;
                 }
-                storeConfig(ContainerRuntime.DOCKER);
+                storeContainerRuntimeInSystemProperty(ContainerRuntime.DOCKER);
                 return ContainerRuntime.DOCKER;
             }
             podmanVersionOutput = getVersionOutputFor(ContainerRuntime.PODMAN);
             podmanAvailable = podmanVersionOutput.startsWith("podman version");
             if (podmanAvailable) {
-                storeConfig(ContainerRuntime.PODMAN);
+                storeContainerRuntimeInSystemProperty(ContainerRuntime.PODMAN);
                 return ContainerRuntime.PODMAN;
             }
+
+            storeContainerRuntimeInSystemProperty(ContainerRuntime.UNAVAILABLE);
+
             if (required) {
                 throw new IllegalStateException("No container runtime was found. "
                         + "Make sure you have either Docker or Podman installed in your environment.");
-            } else {
-                storeConfig(ContainerRuntime.UNAVAILABLE);
-                return ContainerRuntime.UNAVAILABLE;
             }
+
+            return ContainerRuntime.UNAVAILABLE;
         }
     }
 
-    private static ContainerRuntime loadConfig() {
+    private static ContainerRuntime loadContainerRuntimeFromSystemProperty() {
         final String runtime = System.getProperty(CONTAINER_RUNTIME_SYS_PROP);
+
         if (runtime == null) {
             return null;
-        } else if (ContainerRuntime.DOCKER.name().equalsIgnoreCase(runtime)) {
-            return ContainerRuntime.DOCKER;
-        } else if (ContainerRuntime.PODMAN.name().equalsIgnoreCase(runtime)) {
-            return ContainerRuntime.PODMAN;
-        } else if (ContainerRuntime.UNAVAILABLE.name().equalsIgnoreCase(runtime)) {
-            return ContainerRuntime.UNAVAILABLE;
-        } else {
+        }
```

Comment:
Moved that logic to the enum and simplified things a bit. Would appreciate a second pair of eyes to make sure I didn't break something.